### PR TITLE
[idrac_firmware_info] - Enhancement to support iDRAC10

### PR DESCRIPTION
**Pull request by Saksham-Nautiyal**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/idrac_firmware_info.py [hunk 1]
```diff
@@ -104,34 +106,60 @@
...
+
+def get_from_wsman(module):
+    with iDRACConnection(module.params) as idrac:
+        firmware_details = idrac.update_mgr.InstalledFirmware
+        return firmware_details
+
+
+def get_idrac_firmware_info(idrac, module):
+    try:
+        response = idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_URI_10)
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/idrac_firmware_info.py:174]: what if response status code is not 200? handle that scenario.

Comment (Saksham-Nautiyal) [reply plugins/modules/idrac_firmware_info.py:174]: handled

File: plugins/modules/idrac_firmware_info.py [hunk 2]
```diff
@@ -25,7 +25,9 @@
 requirements:
     - "omsdk >= 1.2.488"
```
Comment (rajshekarp87) [inline plugins/modules/idrac_firmware_info.py]:
```suggestion
```

File: plugins/modules/idrac_system_info.py [hunk 3]
```diff
@@ -26,7 +26,9 @@
 requirements:
     - "omsdk >= 1.2.488"
```
Comment (rajshekarp87) [inline plugins/modules/idrac_system_info.py]:
```suggestion
```

File: plugins/modules/idrac_firmware_info.py (hunk 2 above)
Comment (Saksham-Nautiyal) [reply plugins/modules/idrac_firmware_info.py]: done

**Review by sachin-apa — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Pull request merged**